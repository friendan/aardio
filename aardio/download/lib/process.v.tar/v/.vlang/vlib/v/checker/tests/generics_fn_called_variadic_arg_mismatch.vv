fn max<T>(a ...T) T {
	mut max := a[0]
	for item in a[1..] {
		if max < item {
			max = item
		}
	}
	return max
}

fn main() {
	a := max([1, 2, 3, 4])
	println(a)

	b := max<int>([1, 2, 3, 4])
	println(b)
}
