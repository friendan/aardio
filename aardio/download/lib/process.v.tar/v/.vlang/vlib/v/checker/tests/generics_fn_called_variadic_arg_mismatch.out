vlib/v/checker/tests/generics_fn_called_variadic_arg_mismatch.vv:12:11: error: cannot use `[]int` as `int` in argument 1 to `max`
   10 |
   11 | fn main() {
   12 |     a := max([1, 2, 3, 4])
      |              ~~~~~~~~~~~~
   13 |     println(a)
   14 |
vlib/v/checker/tests/generics_fn_called_variadic_arg_mismatch.vv:15:16: error: cannot use `[]int` as `int` in argument 1 to `max`
   13 |     println(a)
   14 |
   15 |     b := max<int>([1, 2, 3, 4])
      |                   ~~~~~~~~~~~~
   16 |     println(b)
   17 | }
